MaterialDesignThemes.Wpf/Extensions.cs

```diff
@@ -41,6 +41,28 @@ public static IEnumerable<DependencyObject> GetVisualAncestory(this DependencyOb
                 yield return leaf;
                 leaf = VisualTreeHelper.GetParent(leaf);
             }
-        } 
+        }
+
+        public static IEnumerable<DependencyObject> GetLogicalAncestory(this DependencyObject leaf)
+        {
+            while (leaf != null)
+            {
+                yield return leaf;
+                leaf = LogicalTreeHelper.GetParent(leaf);
+            }
+        }
+
+        public static bool HasAncestor(this DependencyObject leaf, DependencyObject ancestor)
```

Comment:
I think ```IsDescendantOf``` is a better name.  So usage would be:

```leaf.IsDescendantOf(ancestor) ```

Which makes me think I should rename the original ```IsDescendant``` to ```IsAncestorOf```.